### ISSUE TITLE
Data reported by OpenTelemetry for BatchLoader is incorrect

### ISSUE THREAD

**Issue by cezarypiatekGC** [ChilliCream/graphql-platform#6464]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

`graphql.dataLoader.keys.count` parameter is reported always as 0

Looks like the `_keys collection` is somewhere cleaned before the disposal of DataLoaderBatchScope

![image](https://github.com/ChilliCream/graphql-platform/assets/52414058/75ea2d7e-2686-4569-b765-9df2c1bdd9f8)

### Steps to reproduce

1.Register instrumentation with 
```cs
       .AddInstrumentation(options =>
                {
                    options.Scopes = ActivityScopes.ExecuteHttpRequest | ActivityScopes.DataLoaderBatch;
                    options.RequestDetails = RequestDetails.Id | RequestDetails.Hash | RequestDetails.Operation | RequestDetails.Extensions | RequestDetails.Variables;
                })
```

2. Register batch loader with `AddDataLoader`

### Version

13.4.0

**Comment by michaelstaib** [comment]
The keys are rented ... so you have to copy them if you want to preserve them.

**Comment by michaelstaib** [comment]
damn ... you are right ... its an error :)

**Comment by michaelstaib** [comment]
Fix will be in 13.5-preview.23
